traveler-ios-ui/TravelerKitUI/ViewControllers/BookablePurchaseViewController.swift

```diff
@@ -21,74 +21,25 @@ class BookablePurchaseViewController: UIViewController {
     var bookingContext: BookingContext?
     weak var delegate: BookablePurchaseViewControllerDelegate?
 
-    /// TEMP
-
-    private var passes: [Pass]?
-
-    /// END TEMP
-
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        bookingContext?.addObserver(self)
-        priceLabel.text = bookingContext?.product.price.localizedDescriptionInBaseCurrency
-    }
 
-    deinit {
-        bookingContext?.removeObserver(self)
+        priceLabel.text = bookingContext?.product.price.localizedDescriptionInBaseCurrency
     }
 
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         switch (segue.identifier, segue.destination) {
-        case (_, let vc as BookableConfirmationViewController):
-            vc.modalPresentationStyle = .custom
-            vc.modalPresentationCapturesStatusBarAppearance = true
-            vc.transitioningDelegate = self
-            vc.passes = passes
-            vc.product = bookingContext?.product
-            vc.delegate = self
+        case (_, let vc as BookablePurchaseSelectionViewController):
+            vc.bookingContext = bookingContext
+            vc.errorContext = errorContext
+            vc.delegate = self as? BookablePurchaseSelectionViewControllerDelegate
```

Comment:
just pass self.

We want the compiler to warn us if we havent properly conformed to a protocol